pkg/proxy/roundrobin.go

```diff
@@ -210,14 +236,14 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
-		if !exists || !reflect.DeepEqual(existingEndpoints, validEndpoints) {
+		if !exists || !endpointsAreEqual(existingEndpoints, validEndpoints) {
```

Comment:
`slice.SortStrings([]string) []string` is a weird one in that it sorts in place, and also returns a reference to the slice. It makes it a somewhat strange "util" method and I would worry that it'll trip people up. If you think that's fine, I'll continue on.